### ISSUE TITLE
Change the proof verification functions to return errors instead of bools

### ISSUE THREAD

**Issue by kcalvinalvin** [mit-dci/utreexo#244]
It's hard to tell what the actual errors were because:

1: Error messages are vague
2: Some functions drop some details about errors (ex. IngestBatchProof() in pollardproof.go)

Changing these functions will help debugging easier

**Comment by adiabat** [comment]
Yup, I started doing this in my hashtoknown branch.  Not everything yet but definitely agree.
